pyproject.toml

```
@@ -41,7 +41,7 @@ tomlkit = ">=0.7.0,<1.0.0"
 pexpect = "^4.7.0"
 packaging = "^20.4"
 virtualenv = "^20.4.3"
-keyring = "^21.2.0"
+keyring = ">=21.2.0"
```

```suggestion
keyring = ">=21.2.0,<24.0.0"
```
Might be better to keep an upper limit. Additionally, we should probably update the lock file as well to get 23.x tested.

---

The keyring project is not using semantic versioning so there's no reason to believe that version 24 is a "major" step from version 23.  At their "major" release cadence of less than one month, pip will be reporting errors again next month with this upper limit.

In this case, I think it would be better to leave it open-ended and clamp the range in case there's a breakage.

It may also be worth asking the keyring maintainers to start using semantic versioning.

---

I am honestly on the fence on this. But assuming keyring is not following semver, this makes sense. I had a brief look on their changelog and it seems breaking changes are not highlighted anyway.

Poetry being an application and "expected" installations being in isolated environments, pip warnings should not really impact poetry installs, outside of distro installs. 

TLDR: I agree and we can try the min version see how much pain it actually brings :)